src/Feature.js

```diff
@@ -53,8 +53,59 @@ class Feature {
    * @param {Object} the repo to start a feature in
    * @param {String} branch name to finish feature with
    */
-  static finishFeature() {
-    // TODO
+  static finishFeature(repo, featureName) {
+    if (!repo) {
+      return Promise.reject(new Error('Repo is required'));
+    }
+
+    if (!featureName) {
+      return Promise.reject(new Error('Feature name is required'));
+    }
+
+    let developBranch;
+    let featureBranch;
+    let developCommit;
+    let featureCommit;
+    return Config.getConfig(repo)
+      .then((config) => {
+        const developBranchName = config['gitflow.branch.develop'];
+        const featureBranchName = config['gitflow.prefix.feature'] + featureName;
+
+        return Promise.all(
+          [developBranchName, featureBranchName]
+            .map((branchName) => NodeGit.Branch.lookup(repo, branchName, NodeGit.Branch.BRANCH.LOCAL))
+        );
+      })
+      .then((branches) => {
+        developBranch = branches[0];
+        featureBranch = branches[1];
+        return Promise.all(branches.map((branch) => repo.getCommit(branch.target())));
+      })
+      .then((commits) => {
+        developCommit = commits[0];
+        featureCommit = commits[1];
+        return NodeGit.Merge.commits(repo, developCommit, featureCommit);
+      })
+      .then((index) => {
+        if (!index.hasConflicts()) {
+          index.write();
+          return index.writeTreeTo(repo);
+        }
+        return Promise.reject(`Failed merging ${featureBranch.name()} into ${developBranch.name()}`);
+      })
+      .then((oid) => {
+        const commitMessage = `Merged branch ${featureBranch.name()} into ${developBranch.name()}`;
+        return repo.createCommit(
+          developBranch.name(),
+          repo.defaultSignature(),
+          repo.defaultSignature(),
```

Comment:
You should probably use the same signature object for both of these parameters (they get a timestamp generated, which SHOULD be the same since they are being called right after each other but they could be slightly different).